website/blog/unusual-builds-w-bytes.md

```diff
@@ -0,0 +1,217 @@
+}
+```
+These are typically observed when an action is about to be executed locally, but some of its inputs were produced by a parent action that was executed remotely.
+In such cases, Bazel would typically issue "prefetch" download requests right after the parent action finishes so that the parent action's outputs can be downloaded to the local machine before the child action starts.
+These requests are often annotated with the `prefetcher` action ID and the target ID of the parent action.
+
+We know that in a typical `go_test` rule, there are a few actions involved:
+
+```
+GoCompilePkg
+GoCompilePkgExternal
+=> GoLink
+  => TestRunner
+```
+
+and in this case, we can see that all four actions were executed remotely, eliminating the need for any local download.
+
+# The Investigation
+
+While the original intention of the `prefetcher` requests was to [support locally executed actions](https://github.com/bazelbuild/bazel/commit/ea4ad30d4e49b83b62df0e10b5abe2faadea7582), we found that over time, Bazel has been using this action ID for other purposes.
+For example, if a user were to specify a `--remote_download_regex=` pattern to download specific artifacts, those download requests would also be annotated with the `prefetcher` action ID.
+
+As we digged deeper into the issue, we tried to improve the visibility of Prefetch actions in Bazel with [PR#25040](https://github.com/bazelbuild/bazel/pull/25040), which was backported to Bazel 8.1.0 and 7.5.0.
+This change breaks down the `prefetcher` action ID into "input" downloads and "output" downloads, allowing us to narrow down the code paths that were triggering the unwanted downloads.
+
+Once we had [upgraded to a Bazel version](./blog/bisect-bazel) with this change, we were able to see that the unwanted downloads were coming from the `output` downloads.
+
+# The Bug
+
+We discovered another code path in Bazel that can trigger unexpected downloads: when a blob's TTL expires.
+
+> So... What is a blob's TTL? 🤔
+
+As an artifact-oriented build system, Bazel relies heavily on artifacts existing in its output path to determine if a build is up-to-date.
+If an artifact exists and matches the expected result, then Bazel will skip the action that produces the artifact.
+However, in a remote build with minimal download, Bazel does not have the artifact locally to validate against and must establish some trust that the artifact was created and stored in the Remote Cache.
+
+Introduced in [PR#17639](https://github.com/bazelbuild/bazel/pull/17639), this "trust" is added to Bazel by storing the expected Time To Live (TTL) of the artifact in the Remote Cache.
+When Bazel executes an action remotely, instead of downloading the outputs referenced inside the ActionResult, it instead just stores the references inside its in-process analysis cache (aka. SkyFrame) with an expected TTL that is determined by the flag `--experimental_remote_cache_ttl` (default: 3 hours).
```

Comment:
```suggestion
When Bazel executes an action remotely, instead of downloading the outputs referenced inside the ActionResult, it instead just stores the references inside its in-process analysis cache (aka. Skyframe) with an expected TTL that is determined by the flag `--experimental_remote_cache_ttl` (default: 3 hours).
```